src/bluecherry.c

```diff
@@ -357,15 +361,17 @@ static esp_err_t _bluecherry_coap_rxtx(_bluecherry_msg_t *msg)
                    (1 + (rand() / (RAND_MAX + 1.0)) * 
                    (BLUECHERRY_ACK_RANDOM_FACTOR - 1));
 
-  _bluecherry_opdata.state = BLUECHERRY_STATE_CONNECTED_AWAITING_RESPONSE;
-
   for(uint8_t attempt = 1; attempt <= BLUECHERRY_MAX_RETRANSMITS; ++attempt) {
     _bluecherry_opdata.last_tx_time = time(NULL);
 
+    ESP_LOGD(TAG, "Sending CoAP message (attempt %u, timeout %.1fs)", attempt, timeout);
```

Comment:
I like to keep the library clean of LOGI/D and only include LOGE. This to keep the footprint as small as possible. It's up to the end user to add extra logs in the application layer.